### ISSUE TITLE
`Ixed Value` is a misfeature

### ISSUE THREAD

**Issue by sjshuck** [lens/lens-aeson#43]
What `ix` does on a `Value` is, it filters it as an `Object`, and then looks up a key.  But filtering for objects is an arbitrary decision: there are two possible notions of an index for a JSON value:

- numerical (for arrays)
- textual (for objects, called "names" in the spec)

Granted, people are going to choose the textual notion of indexing a value 99% of the time.  But they can just use `key`, of which `ix @Value` is a type-restricted version.  `ix @Value` doesn't add anything practical, even if it were conceptually sound (which it isn't IMO).

**Comment by RyanGlScott** [comment]
I agree that the `Ixed` instance does assume an `Object`-centric view of JSON, which is perhaps undesirable. I personally wouldn't miss this instance were it to be removed.

@ekmett, you added this instance in 688cf74d8e328f5628b6811ed768e36e07bae63e. Do you have a particular use case for it?

**Comment by ekmett** [comment]
It was mostly a 'hey if you go to use feature x with y and it has a meaningful interpretation it should work' sort of thing. The fact that it is only one half of the object/array split is why there can't be an `At` instance, and if forced to choose, objects are the bulk of what i see folks use json to hold onto/index into rather than arrays.

Is this a perfect instance? no. Is it a legal instance? yes. For me the key distinction (no pun intended) is that removing it doesn't make room for anything better, it just breaks stuff.

**Comment by sjshuck** [comment]
Thanks for the explanation!
